### PR TITLE
add spec for Range#=== to behave like Range#cover? in Ruby 2.7

### PR DESCRIPTION
**Pull request by lxxxvi**
Hello again 👋

From https://github.com/ruby/spec/issues/745 :

> * [ ] Range#=== now uses Range#cover? for String arguments, too (in Ruby 2.6, it was
changed from Range#include? for all types except strings). Bug #15449

To _cover_ that (pun intended), I basically just added this to `Range#===` specs

```ruby
it_behaves_like :range_cover_and_include, :cover?
it_behaves_like :range_cover, :cover?
```

Do you think this is sufficient? 🙈

### CODE REVIEW

File: core/range/case_compare_spec.rb [hunk 1]
```diff
@@ -25,4 +25,9 @@
       (range === RangeSpecs::WithoutSucc.new(2)).should == true
     end
   end
+
+  ruby_version_is "2.7" do
+    it_behaves_like :range_cover_and_include, :cover?
+    it_behaves_like :range_cover, :cover?
```
Comment (eregon) [inline core/range/case_compare_spec.rb:31]: The second argument should be `:===` then

**Review by eregon — COMMENTED** [review]
Sounds good, sorry for the late review.

**Comment by lxxxvi** [comment]
@eregon Thanks for your review!

Interestingly some tests started to fail now (see bottom). Apparently `:===` doesn't work the same as `cover?` when comparing with number ranges. Example

```ruby
# Ruby v2.7.2

(2..5).cover?((3..4)) # => true
(2..5) === (3..4)     # => false
```

I am not sure if this is expected, hence if I understand the [Bug #15449 description](https://bugs.ruby-lang.org/issues/15449) correctly.

Should `:===` work **exactly** like `.cover?` or just particularly for strings? 

What do you think?

### Failing tests


```
Range#=== range argument accepts range argument FAILED
Expected false to be true
/.../core/range/shared/cover.rb:97:in `block (4 levels) in <top (required)>'
/.../core/range/case_compare_spec.rb:5:in `<top (required)>'
```

```ruby
(0..10).send(@method, (3..7)).should be_true
```

---

```
Range#=== range argument supports boundaries of different comparable types FAILED
Expected false to be true
/.../core/range/shared/cover.rb:120:in `block (4 levels) in <top (required)>'
/.../core/range/case_compare_spec.rb:5:in `<top (required)>'
```

```ruby
(0..10).send(@method, (3.1..7.9)).should be_true
```

---

```
Range#=== range argument honors exclusion of right boundary (:exclude_end option) FAILED
Expected false to be true
/.../core/range/shared/cover.rb:132:in `block (4 levels) in <top (required)>'
/.../core/range/case_compare_spec.rb:5:in `<top (required)>'
```

```ruby
(0..10).send(@method, (0..10)).should be_true
```

**Comment by eregon** [comment]
It's expected `(0..10).send(@method, (3..7))` is only true for `cover?` itself.
`===` only has the "does the Range include that element?", not the "is this Range a superset of that Range?" semantics.

So let's move https://github.com/ruby/spec/blob/c8bc53a8e8ac3014f9727297e8784aa04f30cead/core/range/shared/cover.rb#L94-L152 into its own `describe :range_cover_subrange, shared: true do`, that way we only test that for `Range#cover?` itself.

**Comment by lxxxvi** [comment]
@eregon Thank you for your explanation. I added a new commit according your suggestion.

**Review by eregon — APPROVED** [review]
Thank you, looks good!

**Pull request merged**